packages/subscription/src/chainable-subscription.ts

```diff
@@ -12,7 +12,20 @@ export class ChainableSubscription<T,TReturn> implements Subscription<T,TReturn>
     let chain = new ChainableSubscription(DUMMY);
 
     return yield resource(chain, function*() {
-      chain.subscription = yield subscribe(source);
+      chain.subscription = yield rawSubscribe(source);
+      yield;
+    });
+  }
+
+  static *wrap<T, TReturn, R = T>(
+    source: Operation<ChainableSubscription<T, TReturn>>,
+    fn: (inner: ChainableSubscription<T, TReturn>) => ChainableSubscription<R, TReturn>,
+  ): Operation<ChainableSubscription<R, TReturn>> {
+    let chain = new ChainableSubscription<R, TReturn>(DUMMY);
+
+    return yield resource(chain, function*() {
+      let subscription = yield source;
+      chain.subscription = fn(subscription);
       yield;
     });
   }
```

Comment:
The purpose of this is to retain any resources created in the wrapped operation.